### ISSUE TITLE
Train Logistic Regression : Train the model

### ISSUE THREAD

**Issue by kazourak** [farinaleo/ft_dslr#26]
# Train Model

## Description

The aim of the project is, with a logistic regression, to create a model able to predict in witch house a student is, based on his grades in each subject.
The Subject requires to use the gradient descent for the model.

## Acceptance Criteria:
- [x] Load data (cleaned and normalised)
- [x] Separate the characteristic columns (X) and the target variable (y)
- [x] Setup starting param (learning rate and epoch).
- [x] define usefull function (sigmoid, log-cost, gradient)
- [x] Apply the gradient descent

## Useful Links:
[gradient descent](https://en.wikipedia.org/wiki/Gradient_descent)
[logistic regression](https://en.wikipedia.org/wiki/Logistic_regression)
[sigmoid](https://en.wikipedia.org/wiki/Sigmoid_function)

##  Required Math:
- Sigmoid function
$$\sigma = {1\over{1 - e^{-x}}}$$
- Cost (loss) function
$${J(\theta)} = {-{1\over m} \sum_{i=1}^{m} y^i log(h_{\theta}(x^i)) + (1 - y^i) log(1 - h_{\theta}(x^i)) }$$
where:
$$h_{\theta}(x) = g(\theta^{T}x)$$
- partial derivative from loss function:
$${{\partial \over {\partial \theta_j}}J(\theta)} = {{1 \over m} \sum_{i=1}^{m}(h_{\theta}(x^i) - y^i)x^i_j}$$
